.github/workflows/build.yml

```
@@ -0,0 +1,240 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Triggers the workflow on push
+on: [push, pull_request]
+
+jobs:
+  # This workflow contains a single job called "build"
+  build-shared:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      # Cache node_modules folder and bust cache on yarn.lock changes
+      - name: Use cached node_modules
+        uses: actions/cache@v2
+        with:
+          path: "**/node_modules"
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
+      # Use Node.js environment
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v2.1.2
+        with:
+          node-version: 12.x
+
+      - run: yarn install --frozen-lockfile
+      - run: yarn build:shared
+
+      - name: Upload shared build artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: shared-build
+          path: packages/shared/dist
+
+  build-shared-components:
+    needs: build-shared
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      # Cache node_modules folder and bust cache on yarn.lock changes
```

this comment is also a lil misleading because we're pulling from the cache instead of caching? 
```suggestion
      # Get node_modules folder from cache and update dependencies if yarn.lock has changed
```